packages/authorization-server/test/api/authorizationServer.api.test.ts

```diff
@@ -0,0 +1,142 @@
+  platformStateValidationFailed,
+  tokenGenerationStatesEntryNotFound,
+} from "../../src/model/domain/errors.js";
+
+describe("POST /authorization-server/token.oauth2", () => {
+  const clientId = getMockClient().id;
+  const tokenClientKidPK = makeTokenGenerationStatesClientKidPK({
+    clientId,
+    kid: "",
+  });
+
+  const header: InteropJwtHeader = {
+    alg: "RS256",
+    use: "sig",
+    typ: "at+jwt",
+    kid: generateId(),
+  };
+
+  const payload: InteropJwtConsumerPayload = {
+    jti: generateId(),
+    iss: "interop.pagopa.it",
+    aud: ["interop.pagopa.it"],
+    iat: 10,
+    nbf: 100,
+    exp: 100,
+    client_id: clientId,
+    sub: generateId<ClientId>(),
+    purposeId: generateId<PurposeId>(),
+  };
+
+  const token: InteropConsumerToken = {
+    header,
+    payload,
+    serialized: "",
+  };
+
+  const validRequestBody: authorizationServerApi.AccessTokenRequest = {
+    client_id: "e58035ce-c753-4f72-b613-46f8a17b71cc",
+    client_assertion: "valid-jws-token",
+    client_assertion_type:
+      "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+    grant_type: "client_credentials",
+  };
+
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  const makeRequest = (
+    body: authorizationServerApi.AccessTokenRequest = validRequestBody
+  ) =>
+    request(api)
+      .post("/authorization-server/token.oauth2")
+      .set("Content-Type", "application/x-www-form-urlencoded")
+      .send(new URLSearchParams(body as Record<string, string>).toString());
+
+  beforeEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  it("Should return 200 for valid request", async () => {
+    tokenService.generateToken = vi.fn().mockResolvedValue({
+      limitReached: false,
+      token,
+      rateLimiterStatus: {
+        maxRequests: 100,
+        rateInterval: 1,
+        remainingRequests: 10,
+      },
+    });
+
+    const res = await makeRequest();
+
+    expect(res.status).toBe(200);
+    expect(res.body).toEqual({
+      access_token: "",
+      token_type: "Bearer",
+      expires_in: 90,
+    });
+  });
+
+  it.each([
+    {
+      error: tokenGenerationStatesEntryNotFound(tokenClientKidPK),
+      expectedStatus: 400,
+    },
+    {
+      error: clientAssertionRequestValidationFailed(clientId, ""),
+      expectedStatus: 400,
+    },
+    {
+      error: clientAssertionSignatureValidationFailed(clientId, ""),
+      expectedStatus: 400,
+    },
+    {
+      error: clientAssertionValidationFailed(clientId, ""),
+      expectedStatus: 400,
+    },
+    {
+      error: platformStateValidationFailed(""),
+      expectedStatus: 400,
+    },
+  ])(
+    "Should return $expectedStatus for $error.code",
+    async ({ error, expectedStatus }) => {
+      tokenService.generateToken = vi.fn().mockRejectedValue(error);
+      const res = await makeRequest();
+      expect(res.status).toBe(expectedStatus);
+    }
+  );
+
+  it("Should return tooManyRequestsError for 429", async () => {
```

Comment:
maybe rephrase as:
```suggestion
  it("Should return 429 in case the rate limit is reached", async () => {
```
just because I don't actually see `tooManyRequestsError` in the test code